Add instructions for migration from NFS to StorageClass (#2270)
OctopusDeploy/docs: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md

If you need a more reliable storage solution, then you can specify your own `StorageClass`. This `StorageClass` must be capable of `ReadWriteMany` (also known as `RWX`) access mode. 

> Many managed Kubernetes offerings will have provide storage that require little effort to set up. These will be a “provisioner” (named as such as they “provision” storage for a `StorageClass`), which you can then tie to a `StorageClass`. Some examples are listed below:

Reviewer: ```suggestion
Many managed Kubernetes offerings will provide storage that require little effort to set up. These will be a “provisioner” (named as such as they “provision” storage for a `StorageClass`), which you can then tie to a `StorageClass`. Some examples are listed below:
```

- [Longhorn](https://longhorn.io/)
- [Rook (CephFS)](https://rook.io/)
- [GlusterFS](https://www.gluster.org/)

## Migrating from NFS storage to a custom StorageClass

> If you installed the Kubernetes agent using the default NFS storage, and want to use a custom `StorageClass` instead, simply rerun the installation Helm command with specified values for `persistence.storageClassName` and `persistence.size`.

Reviewer: ```suggestion
If you installed the Kubernetes agent using the default NFS storage, and want to change to custom `StorageClass` instead, simply rerun the installation Helm command with specified values for `persistence.storageClassName` and `persistence.size`.
```

Reviewer: Technically they don't need to set the size here unless they want to increase the volume size

The following steps assume your Kubernetes agent is in the `octopus-agent-nfs-to-pv` namespace:

> 1. Take note of the current Helm release and Chart version for your Kubernetes agent by running the following command:

Reviewer: ```suggestion
1. Take note of the current Helm release name and Chart version for your Kubernetes agent by running the following command:
```

   ```bash
   helm list --namespace octopus-agent-nfs-to-pv
   ```
1. The output should look like this:
   - In this example, the release name is `nfs-to-pv` while the chart version is `1.0.1`
   :::figure
   ![Helm list command](/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/kubernetes-agent-helm-list.png)
   :::

> 1. Run the following command (substitute with your own values):

Author: This part looks a bit busy, any suggestions?

Reviewer: There is a code block component that might make this easier to layout?

Example
https://octopus.com/docs/deployments/custom-scripts#scripts-that-block-deployments

Author: I split the steps out into headings - which made it nicer
